[Tan Shao Ning] ip

ningtan11 opened the pull request:
```
Hello, from
 ____        _        
|  _ \\ _   _| | _____ 
| | | | | | | |/ / _ \
| |_| | |_| |   <  __/
|____/ \\__,_|_|\\_\\___|
```
### Your virtual assistant.
##### Has:
- Better memory than ~~you~~ me :sweat_smile:
- Simple, text-based interface
- Open-source `java` code

##### Some features:
- [x] Organised lists -- or at least, as organised as you make them
- [ ] Customisable personality _[To be added]_
- [ ] Deadline reminders _[To be added]_

**Warning:** Duke also features a sassy attitude, and may occasionally refuse to work. 😅 If he does, please report bugs by:

1. Sending an email to the developer, or
2. Leaving a review on [github](https://github.com/ningtan11/ip)

###### Obligatory sample of code ~~to fulfill the fenced code block requirement~~
```java
finally {
                if (isOpen == true) {
                    text = scanner.nextLine().toLowerCase();
                }
}
```

Commit:
Add Gradle support

ningtan11 committed:
level-1

ningtan11 committed:
level-2

ningtan11 committed:
level-3

ningtan11 committed:
level-4

ningtan11 committed:
A-TextUITesting

ningtan11 committed:
level-5

ningtan11 committed:
level-6

seanmanik commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,19 @@
+
+    @Override
+    public String toString() {
+        return "[D]" + super.toString() + " (by: " + time + ")";
```
Excellent formatting here, but one suggestion would be that since type is defined as "[D]" here already, would it be better to return type + super.toString() + ...?

seanmanik commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,153 @@
+        while (open) {
+            try {
+                // Bye
+                if (text.equals("bye") || text.equals("exit")) {
```
In this section, user input is parsed and processed. Would it be better if a Parser class was used to handle this?

seanmanik commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,153 @@
+                    if (list.isEmpty()) {
+                        throw new DukeException("There is nothing in your list yet!");
+                    } else {
+                        System.out.println("Here is your to-do list, Master:");
```
Would this be better done within TaskList, or in a Command class? Also, one other recommendation would be for outputs to be displayed via a Ui class :)

seanmanik commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,153 @@
+    private static ArrayList<Task> list;
+
+    public static void main(String[] args) throws DukeException {
+        list = new ArrayList<>();
```
While this certainly works, it may be better to specify the type that this list would contain. Also, would it be better to have a TaskList class to keep track of Tasks?

seanmanik commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,42 @@
+public class Task {
+    String name;
+    boolean isDone;
+    String type = "[T]";
```
Would this variable be used? If not, it might be better to remove it

seanmanik commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,19 @@
+public class Event extends Task {
```
Excellent use of OOP here, and its great that the "type" variable was correctly used in the toString() method.

seanmanik reviewed (commented):
Naming conventions were followed, and the code was easy to understand. However, one suggestion would be to incorporate a bit more use of OOP principles instead of allowing Duke to handle parsing, UI displays, and task list management. Other than that, keep up the great work :)

drkkjh commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,153 @@
+                                        " as done, Master.");
+                            }
+                        }
+                    }
```
Perhaps these could be extracted into a  separate class that handles the marking of tasks? 😃

drkkjh commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,153 @@
+                        System.out.println("I have added " + newTask.toString() + " to the list, Master.");
+                    } else {
+                        throw new DukeException("I need to know the deadline to add this task to the list, Master.");
+                    }
```
Perhaps these could also be extracted to a ```Parser``` class that handles all the parsing of input from user?

drkkjh commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,153 @@
+                        }
+                        int number = Integer.parseInt(taskNumber) - 1;
+                        if (number >= list.size() || number < 0) {
+                            throw new DukeException("There is no task " + taskNumber + " just yet, Master.");
```
I like that you handled edge cases here 👍

drkkjh commented on src/main/java/ToDo.java:
```diff
@@ -0,0 +1,10 @@
+public class ToDo extends Task {
```
Perhaps you may want to implement a final variable `type` here as in like the other classes `Deadline` and `Event`?

drkkjh reviewed (commented):
Great work overall, excellent use of OOP in general, just a few nits to fix to have a stronger usage of OOP such as extracting out code that parses user input into a ```Parser``` class.

ningtan11 committed:
level-7

ningtan11 committed:
level-8

ningtan11 committed:
A-MoreOOP, A-Packages

ningtan11 committed:
Merge pull request #1 from nus-cs2103-AY2223S1/add-gradle-support

Add Gradle support

ningtan11 committed:
no message

ningtan11 committed:
level-9

ningtan11 committed:
add gradle and javadoc

ningtan11 committed:
Add GUI
Exception handling still needs work, dialog box does not print exception text

ningtan11 committed:
Merge branch 'master' of https://github.com/ningtan11/ip

# Conflicts:
#	build.gradle
#	gradlew

ningtan11 committed:
Fix build.gradle

ningtan11 committed:
Add assert statements

ningtan11 committed:
Add extension to edit tasks

ningtan11 committed:
Fix bugs and improve error handling.

Fixed several off-by-one bugs in MarkCommand, UnmarkCommand, DeleteCommand.

Created independent classes for exceptions that are thrown more than once for neater code.

Changed implementation of FindCommand to use FoundList instead, and return a list of items found that retain the original indices of the items as on the main TaskList.

ningtan11 committed:
Make coffee

Add a flavour command to make coffee, just to hopefully make a user's day.
Give Duke more personality
Fix bugs that arose in parser revamp
First draft of UG

ningtan11 committed:
Fix off-by-one bug in foundlist
Complete UG draft

ningtan11 committed:
Fix formatting issues in UG

ningtan11 committed:
Remove unused methods

ningtan11 committed:
Update AboutUs.md
Add ningtan11.md

ningtan11 committed:
Delete tp-master/docs directory

ningtan11 committed:
no message

ningtan11 committed:
no message

ningtan11 committed:
Add Javadoc comments
